odoo/models.py

```diff
@@ -2407,7 +2407,7 @@ def _read_group_format_result(self, rows_dict, lazy_groupby):
 
                 additional_domain = [(field_name, '=', value)]
 
-                if field.type in ('date', 'datetime'):
+                if field.type in ('date', 'datetime') and (isinstance(value, datetime.date) or isinstance(value, datetime.datetime)):
```

Comment:
`isinstance` can receive tuple of types: => `isinstance(value, (datetime.date, datetime.datetime))`

Sadly, I made a mistake in my proposal, it doesn't take in account `False` value. You can move the extra condition to the next `if` to working as expected.